codetf.schema.json

```diff
@@ -0,0 +1,228 @@
+        }
+      },
+      "required": ["vendor", "tool", "version", "commandLine", "elapsed", "directory"]
+
+    },
+
+    "result": {
+      "type": "object",
+      "properties": {
+        "codemod": {
+          "type": "string",
+          "description": "ID of the transformation that was applied"
+        },
+        "summary": {
+          "type": "string",
+          "description": "A short phrase describing the changes made"
+        },
+        "description": {
+          "type": "string",
+          "description": "A longer description of the changes made"
+        },
+        "detectionTool": {
+          "description": "Metadata about the tool that detected the issue",
+          "$ref": "#/definitions/detectionTool"
+        },
+        "references": {
+          "type": "array",
+          "items": { "$ref": "#/definitions/reference" }
+        },
+        "properties": {
+          "type": "object",
+          "description": "Vendor-specific properties to help storytelling",
+          "additionalProperties": true
+        },
+        "failedFiles": {
+          "type": "array",
+          "description": "A set of file paths for files that the tool failed to parse or transform",
+          "items": { "type": "string" }
+        },
+        "changeset": {
+          "type": "array",
+          "description": "The changes for a given codemod",
+          "items": { "$ref": "#/definitions/changeset" },
+          "minItems": 0
+        }
+      },
+      "required": ["codemod", "summary", "description", "changeset"]
+    },
+
+    "sarif": {
+      "type": "object",
+      "properties": {
+        "artifact": {
+          "type": "string",
+          "description": "The file containing the SARIF or other input"
+        },
+        "sha1": {
+          "type": "string",
+          "description": "SHA1 hash representing the artifact"
+        }
+      },
+      "required": ["artifact", "sha1"]
+    },
+
+    "reference": {
+      "type": "object",
+      "properties": {
+        "url": {
+          "type": "string",
+          "description": "The URL of the reference"
+        },
+        "description": {
+          "type": "string",
+          "description": "A description of the reference"
+        }
+      },
+      "required": ["url"]
+    },
+
+    "changeset": {
+      "type": "object",
+      "properties": {
+        "path": {
+          "type": "string",
+          "description": "The path of the file (relative to the directory) that was changed"
+        },
+        "diff": {
+          "type": "string",
+          "description": "The unified diff format of the changes made to the file"
+        },
+        "changes": {
+          "type": "array",
+          "items": { "$ref": "#/definitions/change" },
+          "minItems": 1
+        }
+      },
+      "required": ["path", "diff", "changes"]
+    },
+
+    "change": {
+      "type": "object",
+      "properties": {
+        "lineNumber": {
+          "type": "string",
```

Comment:
number

Reply:
I agree 💯 but I think our tools generate this as a string since that's how its described in the current version of the spec and I'm hesitant to change it for fear of breaking platform integration.

Reply:
I ran `check-jsonschema` on CodeTF that I pulled out of the codemod orchestrator, and it broke on this property

```
  /Users/jgilday/Downloads/results.codetf-8.json::$.results[1].changeset[2].changes[0].lineNumber: 78 is not of type 'string'
```

Reply:
Since Python is generating strings for these fields I'm going to allow both for now. But I think I should change the Python behavior and eventually remove this possibility from the spec.